### ISSUE TITLE
Per-project file/folder associations

### ISSUE THREAD

**Issue by deakjahn** [AtomMaterialUI/a-file-icon-idea#463]
Is it possible to get per-project associations? VS Code makes it easy with its settings structure but in Android Studio and other related IDEs? Even if it would require manual editing of XML settings files... I often have specific needs in different projects that I don't want to pollute a global settings table with, or can even contradict each other in different projects.

**Comment by mallowigi** [comment]
I have it in mind, it will come in a future release, but I don't know when.

**Comment by deakjahn** [comment]
Yep, that's why I said that something as simple as an `.atom_icons` source file, if present, with a somewhat documented XML format (could be as simple as a single example file) could be just as good. All your users are programmers, by definition, after all.. :-)

**Comment by mallowigi** [comment]
Yes but you'd still need to create the PersistentStateComponent either way.

Anyway if you want to tackle this, be my guest, I am already in the middle of another feature and I don't have a lot of time for side-projects...

**Comment by edclement** [comment]
> Is it possible to get per-project associations?

@deakjahn - [Extra Icons](https://plugins.jetbrains.com/plugin/11058-extra-icons) plugin allows overriding icons per project.

@mallowigi - Atom Material Icons is a great plugin btw. Not trying to push people to use something different. Just noting that plugin has the per-project functionality.

**Comment by mallowigi** [comment]
I know, allow adding custom file associations per project is on the roadmap, it shouldn't be that hard, however I have other priorities at the moment. Perhaps for next release

**Comment by mallowigi** [comment]
Done. Thanks AI ;)
